src/components/Modals/AssetsSelect/index.tsx

```diff
@@ -1,81 +1,183 @@
+  const [nonCollateralSelected, setNonCollateralSelected] = useState<RowSelectionState>({})
 
   useEffect(() => {
-    const selectedAssets = assets.filter((_, index) => selected[index])
+    let newSelectedDenoms: string[]
+
+    if (Array.isArray(tableData)) {
+      newSelectedDenoms = assets
+        .filter((asset, index) =>
+          tableData.some((row, idx) => row.asset.denom === asset.denom && whitelistedSelected[idx]),
+        )
+        .map((asset) => asset.denom)
+    } else {
+      const whitelistedAssets = assets.filter((asset) =>
+        tableData.whitelistedData.some(
+          (row, index) => row.asset.denom === asset.denom && whitelistedSelected[index],
+        ),
+      )
+      const nonCollateralAssets =
+        nonCollateralTableAssets?.filter((asset) =>
+          tableData.nonCollateralData.some(
+            (row, index) => row.asset.denom === asset.denom && nonCollateralSelected[index],
+          ),
+        ) || []
+
+      newSelectedDenoms = [...whitelistedAssets, ...nonCollateralAssets]
+        .sort((a, b) => a.symbol.localeCompare(b.symbol))
+        .map((asset) => asset.denom)
+    }
 
-    const newSelectedDenoms = selectedAssets
-      .sort((a, b) => a.symbol.localeCompare(b.symbol))
-      .map((asset) => asset.denom)
     if (
       selectedDenoms.length === newSelectedDenoms.length &&
       newSelectedDenoms.every((denom) => selectedDenoms.includes(denom))
     )
       return
     onChangeSelected(newSelectedDenoms)
-  }, [selected, props, assets, selectedDenoms, onChangeSelected])
+  }, [
+    whitelistedSelected,
+    nonCollateralSelected,
+    tableData,
+    selectedDenoms,
+    onChangeSelected,
+    assets,
+    nonCollateralTableAssets,
+  ])
+
+  if (!nonCollateralTableAssets) {
+    return (
+      <Table
+        title='Assets'
+        columns={columns}
+        data={tableData as AssetTableRow[]}
+        initialSorting={sorting}
+        onSortingChange={setSorting}
+        setRowSelection={setWhitelistedSelected}
+        selectedRows={whitelistedSelected}
+        hideCard
+      />
+    )
+  }
 
   return (
-    <Table
-      title='Assets'
-      columns={columns}
-      data={tableData}
-      initialSorting={[{ id: isBorrow ? 'asset.borrowRate' : 'value', desc: !isBorrow }]}
-      setRowSelection={setSelected}
-      selectedRows={selected}
-      hideCard
-    />
+    <>
+      <Table
+        title='Assets that can be used as collateral'
+        columns={columns}
+        data={createTableData(assets)}
+        initialSorting={sorting}
+        onSortingChange={setSorting}
+        setRowSelection={setWhitelistedSelected}
+        selectedRows={whitelistedSelected}
+        hideCard
+        titleComponent={
+          <div className=''>
+            <Text
+              size='xs'
+              className='p-4 font-semibold bg-black/20 text-white/60 border-t border-b border-white/10 w-full'
```

Comment:
Prevent double border
```suggestion
              className='p-4 font-semibold bg-black/20 text-white/60 border-b border-white/10 w-full'
```